include/stout/flags/flags.cc

```diff
@@ -663,6 +670,52 @@ void Parser::Parse(const std::vector<ArgumentInfo>& values) {
 
 ////////////////////////////////////////////////////////////////////////
 
+void Parser::SetFieldMessageOrAggregateErrors(
+    const std::string& value,
+    const std::string& name,
+    const google::protobuf::FieldDescriptor* field,
+    google::protobuf::Message* message,
+    std::set<std::string>& errors) {
+  // Parse the value using an error collector that aggregates the
+  // error for us to print out later.
+  struct ErrorCollector : public google::protobuf::io::ErrorCollector {
+    void AddError(
+        int /* line */,
+        int /* column */,
+        const std::string& message) override {
+      error += message;
+    }
+
+    void AddWarning(
+        int line,
+        int column,
+        const std::string& message) override {
+      // For now we treat all warnings as errors.
+      AddError(line, column, message);
+    }
+
+    std::string error;
+  } error_collector;
+
+  google::protobuf::TextFormat::Parser text_format_parser;
+  text_format_parser.RecordErrorsTo(&error_collector);
+  if (!text_format_parser.ParseFieldValueFromString(
+          value,
+          field,
+          &GetMessageForField(*field))) {
```

Comment:
Should we pass in `message` here instead of calling `GetMessageForField`?

If not, the `message` arg looks unused and we can remove it. (But do we want that? Does everyone just want to rely on this method calling `GetMessageForField`?)